livekit-agents/livekit/agents/pipeline/pipeline_agent.py

```diff
@@ -883,24 +883,25 @@ async def _execute_function_calls() -> None:
             if interrupted:
                 collected_text += "..."
 
-            msg = ChatMessage.create(text=collected_text, role="assistant")
-            self._chat_ctx.messages.append(msg)
+            if collected_text:
```

Comment:
When can this happens? I'm more worried about the cause than fixing the scenario where it is indeed empty

Reply:
This happens when agent finish speaking  tool call's response. we would also get an extra empty transcript. [screenshot](https://github.com/livekit/agents/pull/1148#issue-2702435425)

Can there be a possiblity when llm response is just empty?

Reply:
Ah right, I think we should add new arguments to the agent_speech_committed event. I think it is a valid usecase to have an empty content but having a function calls

Reply:
> Ah right, I think we should add new arguments to the agent_speech_committed event.

@theomonnom wouldn't that break existing callbacks? I'm not sure if there's value in it because function calls are supposed to be handled by the framework

Reply:
I will think more about that alongside the new IO proposal

Reply:
Let's put this PR in standby for now